File: gitea_repos_master/whatsapp/whatsapp-baileys-api/src/logic/thread/exception_handler.ts
import * as process from 'process';
import { MongoSessions } from '../mongo/client';
import KeyDBQueue from '../keydb/keydb';
import { KeyDBResponseBuilder } from '../keydb/response_builder';
import logger from '../logger/winston';
import { Session } from 'own';
import { HealthCheck } from '../livenessprobe/health_check';
import { safe } from '../utils/func';

export async function noData(
  mongo: MongoSessions,
  keydb: KeyDBQueue,
  session: Session,
  payload: string
) {
  const sessionSuccess = mongo.sessionSuccess.bind(mongo);
  await Promise.all([
    keydb.setAnswer(payload, KeyDBResponseBuilder.empty()),
    safe(sessionSuccess, false, 0, session, 1, 3)
  ]);
  HealthCheck.checkpoint();
  logger.info('Success task: user not found');
}

export async function accountLocked(
  mongo: MongoSessions,
  keydb: KeyDBQueue,
  session: Session,
  payload: string
) {
  await keydb.setAnswer(
    payload,
    KeyDBResponseBuilder.error('Учетная запись заблокирована', 506)
  );
  process.exit(1);
}

export async function returnToQueue(
  mongo: MongoSessions,
  keydb: KeyDBQueue,
  session: Session,
  payload: string
) {
  await keydb.setAnswer(
    payload,
    KeyDBResponseBuilder.error('Произошла ошибка возврата в очередь', 500)
  );
}

File: gitea_repos_master/pochta/pochta-search-api/src/logic/thread/exception_handler.py
import asyncio
from collections import OrderedDict

from isphere_exceptions.session import SessionBlocked
from isphere_exceptions.success import NoDataEvent
from worker_classes.keydb.response_builder import KeyDBResponseBuilder


async def normal(exception, mongo, kdbq, session, payload, api):
    pass


async def no_data_error(exception, mongo, kdbq, session, payload, api):
    await asyncio.gather(
        kdbq.set_answer(payload, KeyDBResponseBuilder.empty()),
        mongo.session_success(session),
    )


async def session_blocked(exception, mongo, kdbq, session, payload, api):
    await asyncio.gather(
        kdbq.set_answer(
            payload,
            KeyDBResponseBuilder.error(exception),
        ),
        mongo.session_block(session),
    )


async def base_exception(exception, mongo, kdbq, session, payload, api):
    await kdbq.set_answer(
        payload,
        KeyDBResponseBuilder.error(exception),
    )


order_exceptions = OrderedDict(
    [
        (None, normal),
        (NoDataEvent, no_data_error),
        (SessionBlocked, session_blocked),
        (Exception, base_exception),
    ]
)
